src/components/CategoriesFilters/CategoriesFilters.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect } from 'react';
+import { useAxios } from '../../hooks/useAxios';
+import styles from './CategoriesFilters.module.css';
+import { useGlobalContext } from '../../context/Provider';
+
+function CategoriesFilters() {
+    const [selectedCategories, setSelectedCategories] = useState([]);
+    const { filterProducts, clearFilter } = useGlobalContext();
+
+    const { response, loading } = useAxios({
+        method: 'GET',
+        url: 'product-categories.json',
+    });
+
+    useEffect(() => {
+        return () => {
+            clearFilter();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const handleFilter = (e) => {
+        const val = e.target.value.toLowerCase();
+        const newSelected = [...selectedCategories];
+
+        if (selectedCategories.includes(val)) {
+            let tagIndex = newSelected.indexOf(val);
+            newSelected.splice(tagIndex, 1);
+        } else {
+            newSelected.push(val);
+        }
+        console.log(newSelected);
+        setSelectedCategories(newSelected);
+        console.log(selectedCategories);
+        filterProducts(newSelected);
+    };
```

Comment:
this code looks good but the variable `newSelected` is defined as `const` and the values are modified in the following lines.

```suggestion
    const handleFilter = (e) => {
        const value = e.target.value.toLowerCase();
        const newSelected = selectedCategories.includes(value)
          ? selectedCategories.filter(categorie => categorie !== value)
          : [...selectedCategories, value];

        setSelectedCategories(newSelected);
        filterProducts(newSelected);
    };
```